### PR TITLE
Create a Simple Adapter Test For AWSSQS Target

### PR DESCRIPTION
**Pull request by JeffNeff**

### CODE REVIEW

File: pkg/targets/adapter/awssqstarget/adapter_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	for name, tc := range testCases {
+		//nolint:scopelint
+		t.Run(name, func(t *testing.T) {
+			sqsCli := &standardMockSQSClient{
+				failSendMessage: tc.failSendMessage,
+			}
+			a := adapter{
+				logger:           loggingtesting.TestLogger(t),
+				sqsClient:        sqsCli,
+				awsArnString:     "arn:aws:kinesis:us-east-1:123456789012:stream/foo",
```
Comment (antoineco) [inline pkg/targets/adapter/awssqstarget/adapter_test.go:74]: Kinesis?

File: pkg/targets/adapter/awssqstarget/adapter_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+// standardMockSQSClient is a mocked SQS client which returns a standard set of
+// responses and never errors.
+type standardMockSQSClient struct {
+	sqsiface.SQSAPI
+	sync.Mutex
+	failSendMessage bool
+}
+
+func (c *standardMockSQSClient) SendMessage(*sqs.SendMessageInput) (*sqs.SendMessageOutput, error) {
```
Comment (antoineco) [inline pkg/targets/adapter/awssqstarget/adapter_test.go:119]: To make the test more useful than "failed / didn't fail", we could make the mock client record the input parameter here, and assert that it conforms to what the target should do.

Some inspiration:
https://github.com/triggermesh/triggermesh/blob/54652bfd317467ece28dcf1f6f6a3269d1b94a3e/pkg/targets/adapter/splunktarget/adapter_test.go#L39

**Review by antoineco — CHANGES_REQUESTED** [review]
A good start.
The test needs a few improvements to become really useful (e.g. we should at the very least assert the generated data).

File: pkg/targets/adapter/awssqstarget/adapter_test.go [hunk 3]
```diff
@@ -96,9 +113,13 @@ func makeARN(resource string) arn.ARN {
 type standardMockSQSClient struct {
 	sqsiface.SQSAPI
 	sync.Mutex
```
[the remark this answers is not in the document]
Comment (JeffNeff) [reply pkg/targets/adapter/awssqstarget/adapter_test.go:115]: yes. its copy pasta from your source test.

File: pkg/targets/adapter/awssqstarget/adapter_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+			failSendMessage:  false,
+			expectedResponse: "",
+			expectedEvent:    expectedSucessfulEvent,
+		},
+		"Failure request": {
+			inEvent:          newEvent(t),
+			failSendMessage:  true,
+			expectedResponse: expectedFailureResponse,
+			expectedEvent:    "",
+		},
```
[the remark this answers is not in the document]
Comment (JeffNeff) [reply pkg/targets/adapter/awssqstarget/adapter_test.go:62]: That would greatly complicate the tests.. as we would have to retrieve the message from a Queue to verify this. If we want it, i will do it tho.

File: pkg/targets/adapter/awssqstarget/adapter_test.go [hunk 5]
```diff
@@ -61,8 +76,10 @@ func TestAdapter(t *testing.T) {
 				discardCEContext: false,
 			}
 			event, response := a.dispatch(tc.inEvent)
-			assert.Equal(t, expectedResponse, string(event.DataEncoded))
-			assert.Equal(t, "", response.Error())
+			if event != nil {
```
[the remark this answers is not in the document]
Comment (JeffNeff) [reply pkg/targets/adapter/awssqstarget/adapter_test.go:79]: Im sorry... but i dont understand :/

[the remark this answers is not in the document]
Comment (antoineco) [reply pkg/targets/adapter/awssqstarget/adapter_test.go:109]: You don't check what your test actually expects an event or not, you simply check for `nil` and assumes something based on that.

The correct approach should read as:
- if a expect an event but got `nil`, something is wrong
- if I don't expect an event but got one, something is wrong

**not**

- if I got an event, I probably should have gotten an event

File: pkg/targets/adapter/awssqstarget/adapter_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (cab105) [reply pkg/targets/adapter/awssqstarget/adapter_test.go:95]: Yup

**Review by antoineco — CHANGES_REQUESTED** [review]
The input recorder that was just added isn't even used, so I'm not certain what feedback you want to hear by asking another review from me.

We should record calls to SendMessage by appending to it, then assert that calls to SendMessage contained messages in the expected shape. That's it.

**Comment by antoineco** [comment]
@JeffNeff the **number** of created message is asserted, but this doesn't tell us anything about the SQS message generated by the application.
Enforcing expectations on the data itself is the single most important thing we want to get from this test (because that's what the adapter is for), otherwise we can rename it to "Test that we are sending only one request pretty per incoming event".

I'm thinking about things like
- checking that the data is passed unmodified to the message body
- if CloudEvents context attributes should be propagated to the attributes of the SQS message (not sure if that was part of the requirement when we wrote the target), assert that they are
- etc.

In short, look at what the adapter's code seems to be doing with the incoming event, and ensure the outgoing message is conformant.
